src/main/groovy/org/edx/jenkins/dsl/AnalyticsConstants.groovy

```diff
@@ -0,0 +1,104 @@
+package org.edx.jenkins.dsl
+
+class AnalyticsConstants {
+
+    public static def common_log_rotator = { extraVars ->
+        return {
+            daysToKeep(30)
+        }
+    }
+
+    public static def common_multiscm = { extraVars ->
+        return {
+            git {
+                remote {
+                    url('$TASKS_REPO')
+                    branch('$TASKS_BRANCH')
+                }
+                extensions {
+                    relativeTargetDirectory('analytics-tasks')
+                    pruneBranches()
+                    cleanAfterCheckout()
+                }
+            }
+            git {
+                remote {
+                    url('$CONFIG_REPO')
+                    branch('$CONFIG_BRANCH')
+                }
+                extensions {
+                    relativeTargetDirectory('analytics-configuration')
+                    pruneBranches()
+                    cleanAfterCheckout()
+                }
+            }
+        }
+    }
+
+    public static def common_parameters = { extraVars, env=[:] ->
+        return {
+            stringParam('CLUSTER_NAME', env.get('CLUSTER_NAME', extraVars.get('CLUSTER_NAME')), 'Name of the EMR cluster to use for this job.')
+            stringParam('CONFIG_BRANCH', '$ANALYTICS_CONFIGURATION_RELEASE', 'e.g. tagname or origin/branchname, or $ANALYTICS_CONFIGURATION_RELEASE')
+            stringParam('CONFIG_REPO', 'git@github.com:edx/edx-analytics-configuration.git', '')
+            stringParam('EMR_MASTER_INSTANCE_TYPE', extraVars.get('EMR_MASTER_INSTANCE_TYPE', 'm4.2xlarge'), 'EC2 Instance type used for master.')
+            stringParam('EMR_WORKER_INSTANCE_TYPE_1', extraVars.get('EMR_WORKER_INSTANCE_TYPE_1', 'm4.2xlarge'), 'EC2 instance type used by workers.')
+            stringParam('EMR_WORKER_INSTANCE_TYPE_2', extraVars.get('EMR_WORKER_INSTANCE_TYPE_2', 'm4.4xlarge'), 'EC2 instance type used by workers.')
+            textParam('EMR_HADOOP_ENV_CONFIG', extraVars.get('EMR_HADOOP_ENV_CONFIG'), 'EMR Hadoop env configuration.')
+            textParam('EMR_MAPRED_SITE_CONFIG', extraVars.get('EMR_MAPRED_SITE_CONFIG'), 'EMR mapred-site configuration')
+            textParam('EMR_YARN_SITE_CONFIG', extraVars.get('EMR_YARN_SITE_CONFIG'), 'EMR yarn-site configuration.')
+            textParam('EXTRA_VARS', extraVars.get('EMR_EXTRA_VARS'), $/Extra variables to pass to the EMR provision/terminate ansible playbook.
```

Comment:
nit: This is why I was having trouble with the naming of extraVars elsewhere.    Are they really extra, or are they really 'allVars'?    It made sense when they were just the vars specified for the specific job, but the extraVars here are actually equal to "commonConfigMap + extraVars".